src/main/scala/org/broadinstitute/dsde/workbench/sam/api/SecurityDirectives.scala

```diff
@@ -3,48 +3,99 @@ package org.broadinstitute.dsde.workbench.sam.api
-                  ErrorReport(StatusCodes.NotFound, s"Resource ${resource.resourceTypeName.value}/${resource.resourceId.value} not found")))
-            } else {
-              Directives.failWith(
-                new WorkbenchExceptionWithErrorReport(ErrorReport(
-                  StatusCodes.Forbidden,
-                  s"You may not perform any of ${requestedActions.mkString("[", ", ", "]").toString.toUpperCase} on ${resource.resourceTypeName.value}/${resource.resourceId.value}"
-                )))
-            }
-          }
+          val forbiddenErrorMessage = s"You may not perform any of ${requestedActions.mkString("[", ", ", "]").toUpperCase} on ${resource.resourceTypeName.value}/${resource.resourceId.value}"
+          determineErrorMessage(resource, userId, forbiddenErrorMessage, samRequestContext)
         }
       }
     }
 
+  /**
+    * in the case where we don't have the required action, we need to figure out if we should return
+    * a Not Found (you have no access) vs a Forbidden (you have access, just not the right kind)
+    */
+  private def determineErrorMessage(resource: FullyQualifiedResourceId, userId: WorkbenchUserId, forbiddenErrorMessage: String, samRequestContext: SamRequestContext) = {
+    // in the case where we don't have the required action, we need to figure out if we should return
+    // a Not Found (you have no access) vs a Forbidden (you have access, just not the right kind)
+    onSuccess(policyEvaluatorService.listResourceAccessPoliciesForUser(resource, userId, samRequestContext)) { policies =>
+      if (policies.isEmpty) {
+        Directives.failWith(
+          new WorkbenchExceptionWithErrorReport(
+            ErrorReport(StatusCodes.NotFound, s"Resource ${resource.resourceTypeName.value}/${resource.resourceId.value} not found")))
+      } else {
+        Directives.failWith(
+          new WorkbenchExceptionWithErrorReport(ErrorReport(
+            StatusCodes.Forbidden,
+            forbiddenErrorMessage
+          )))
+      }
+    }
+  }
+
   private def hasPermissionOneOf(resource: FullyQualifiedResourceId, actions: Iterable[ResourceAction], userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] =
     actions.toList.existsM(policyEvaluatorService.hasPermission(resource, _, userId, samRequestContext))
+
+  private def hasParentPermissionOneOf(resource: FullyQualifiedResourceId, newParent: Option[FullyQualifiedResourceId], actions: Iterable[ResourceAction], userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] = {
+    val parentIO = newParent match {
+      case Some(_) => IO.pure(newParent)
+      case None => resourceService.getResourceParent(resource, samRequestContext)
+    }
+
+    parentIO.flatMap {
+      case Some(resourceParent) => hasPermissionOneOf(resourceParent, actions, userId, samRequestContext)
+      case None =>
+        // there is no parent so permission is granted
+        IO.pure(true)
```

Comment:
why do we return true if no parent exists here? It feels weird since we're checking if the user has some permission on the parent. If the parent doesn't exist they wouldn't be able to make changes on the parent right?

Reply:
If there is no parent this effectively disables the parent permission check which seems appropriate.

We hit this case in requiring `remove_child` only at the moment. If there is not parent `remove_child` should be allowed.